inst/include/child_model_simulation.hpp

```diff
@@ -1327,12 +1328,146 @@ void calc_art_initiates(int time_step,
   }// end ag
 
   for (int ag = 0; ag < 4; ++ag) {
-    if (state_next.children.hc_art_init(ag) > (intermediate.children.unmet_need(ag) ))
-      state_next.children.hc_art_init(ag) = (intermediate.children.unmet_need(ag));
+    if (state_next.children.hc_art_init(ag) > (intermediate.children.unmet_need(ag)+intermediate.children.on_art(ag) * cpars.hc_art_ltfu(time_step)))
+      state_next.children.hc_art_init(ag) = (intermediate.children.unmet_need(ag)+intermediate.children.on_art(ag) * cpars.hc_art_ltfu(time_step));
   }// end ag
 
+
+}
+
+template<typename ModelVariant, typename real_type>
+void calc_art_ltfu(int time_step,
+                   const Parameters<ModelVariant, real_type> &pars,
+                   const State<ModelVariant, real_type> &state_curr,
+                   State<ModelVariant, real_type> &state_next,
+                   IntermediateData<ModelVariant, real_type> &intermediate) {
+  static_assert(ModelVariant::run_child_model,
+                "calc_art_ltfu can only be called for model variants where run_child_model is true");
+  constexpr auto ss = StateSpace<ModelVariant>().base;
+  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
+  const auto cpars = pars.children.children;
+
+  for (int s = 0; s <ss.NS; ++s) {
+    for (int a = 0; a < pars.base.options.p_idx_fertility_first; ++a) {
+      for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
+
+        if (a < hc_ss.hc2_agestart) {
+          intermediate.children.hc_hiv_total(hd, a, s) += state_next.children.hc1_hiv_pop(hd, cat, a, s);
+        } else if (hd < (hc_ss.hc2DS)) {
```

Comment:
this is to make sure that we go to the correct number of CD4 categories for the age group. 

So it the age is less than 5, go through all 7 CD4 categories. If age is older than 5 then only go through 6 CD4 categories. This is because 5-14 year olds have one less CD4 category than under five year olds.

Reply:
Any suggestions on a clearer way to do this are welcome, for now I will add it to my documentation list.

Reply:
I would probably make it separate loops to DS determined by value of `a` rather than specifying that as an `else if`